### ISSUE TITLE
Raw return: {  "error": "listener name options not found"}

### ISSUE THREAD

**Issue by DigitalCorrosion** [tristandostaler/EmpireGUI#10]
`> Raw return: {  "listeners": [    {      "ID": 1,       "listener_category": "client_server",       "listener_type": null,       "module": "http",       "name": "http",       "options": {        "BindIP": {          "Description": "The IP to bind to on the control server.",           "Required": true,           "Value": "0.0.0.0"        },         "CertPath": {          "Description": "Certificate path for https listeners.",           "Required": false,           "Value": ""        },         "DefaultDelay": {          "Description": "Agent delay/reach back interval (in seconds).",           "Required": true,           "Value": 5        },         "DefaultJitter": {          "Description": "Jitter in agent reachback interval (0.0-1.0).",           "Required": true,           "Value": 0.0        },         "DefaultLostLimit": {          "Description": "Number of missed checkins before exiting",           "Required": true,           "Value": 60        },         "DefaultProfile": {          "Description": "Default communication profile for the agent.",           "Required": true,           "Value": "/admin/get.php,/news.php,/login/process.php|Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"        },         "Host": {          "Description": "Hostname/IP for staging.",           "Required": true,           "Value": "http://192.168.23.128:80"        },         "KillDate": {          "Description": "Date for the listener to exit (MM/dd/yyyy).",           "Required": false,           "Value": ""        },         "Launcher": {          "Description": "Launcher string.",           "Required": true,           "Value": "powershell -noP -sta -w 1 -enc "        },         "Name": {          "Description": "Name for the listener.",           "Required": true,           "Value": "http"        },         "Port": {          "Description": "Port for the listener.",           "Required": true,           "Value": 80        },         "Proxy": {          "Description": "Proxy to use for request (default, none, or other).",           "Required": false,           "Value": "default"        },         "ProxyCreds": {          "Description": "Proxy credentials ([domain\\]username:password) to use for request (default, none, or other).",           "Required": false,           "Value": "default"        },         "ServerVersion": {          "Description": "Server header for the control server.",           "Required": true,           "Value": "Microsoft-IIS/7.5"        },         "SlackChannel": {          "Description": "The Slack channel or DM that notifications will be sent to.",           "Required": false,           "Value": "#general"        },         "SlackToken": {          "Description": "Your SlackBot API token to communicate with your Slack instance.",           "Required": false,           "Value": ""        },         "StagerURI": {          "Description": "URI for the stager. Must use /download/. Example: /download/stager.php",           "Required": false,           "Value": ""        },         "StagingKey": {          "Description": "Staging key for initial agent negotiation.",           "Required": true,           "Value": "_Bi4>HXK@(|6s*u{:JW3NyMvI}~.?F)8"        },         "UserAgent": {          "Description": "User-agent string to use for the staging request (default, none, or other).",           "Required": false,           "Value": "default"        },         "WorkingHours": {          "Description": "Hours for the agent to operate (09:00-17:00).",           "Required": false,           "Value": ""        }      }    }  ]}
> https://192.168.110.129:1337/api/listeners/options?token=wk8915rmxuuvshqqutbq5nmao2m76opwv4epqe67
> GET
> Raw return: {  "error": "listener name options not found"}`

Get this when running EmpireGUI on the latest release of Empire (- Version 2.3 Master Release
).

**Comment by tristandostaler** [comment]
Hum did not work on this project for a long time now. It might be outdated compared to the new version. When the version 2 came out, I didn't modify it because there was some problems with Empire's REST API so I was waiting for it to be fixed.

If you want, you can totally work on this project, i'll gladly accept your pull requests. It should be easy to fix, something like changing the way the rest api is called (the end point probably).

Thanks for this!

**Comment by DigitalCorrosion** [comment]
yeah unfortunately i am not a coder lol. i can use the comand line like a
wiz and wanted to simplify it by using the GUI. Have any motivation to
update it at all or something that you threw in the back burner? There was
a web GUI but it doesn't work past the initial release of empire sadly.

**Comment by tristandostaler** [comment]
Nahh I will definetly work on it sometime in the future. I have a few features that I would like to add but I don't think I'll have the time. But I will probably fix bugs. I'll keep you posted here if I modify it but it might not be too soon (pretty busy else where ahaha)
The webgui is ok but it's pretty much like using the command line, it doesn't add any new features to simplify everything.

In the features I would like to add I have a few ideas like "macro" of some kind to automate some process (like start a command when an agent comes in, or multiple commands).

**Comment by DigitalCorrosion** [comment]
Yeah, your gui i liked much more than the webui. Be awesome to see it be
integrated again. I would even donate to push yeah a bit :)

**Comment by tristandostaler** [comment]
Well that could be cool! The one thing that would help me the most is to find a few motivated people to help me push my ideas forward. I am pretty sure that donation could motivate them ahaha

**Comment by DigitalCorrosion** [comment]
whats it coded in again? haveny messed with it in a bit. I havr some coding
buddies

**Comment by tristandostaler** [comment]
It's all in java and the modification for this specific bug would all be in the backend part.
